### PR TITLE
Rename coordinate system from incorrect CORSIKA to Ground coordinates; increase test coverage

### LINKED ISSUE #658: Wrong naming of coordinate systems in `print_array_elements`
The naming of the coordinate system `CORSIKA` is not correct in `print_array_elements` and classes associated with it.

- CORSIKA coordinate systems are aligned with geomagnetic North
- this coordinate system in simtools is aligned with geographic North (the difference is handled in CORISKA using the `ARRANG` parameter)

Find a better naming "ground or simtools coordinates?" and replace it accordingly in the code.

### PR DESCRIPTION
**Pull request by GernotMaier**
The naming of the coordinate system `CORSIKA` is not correct in `print_array_elements` and classes associated with it.

- CORSIKA coordinate systems are aligned with geomagnetic North
- this coordinate system in simtools is aligned with geographic North (the difference is handled in CORISKA using the `ARRANG` parameter)

Renamed the coordinate system to "ground", as discussed and agreed in #658.

Added additional unit tests to cover all important cases to be tested.

Closes #658

### CODE REVIEW

File: tests/unit_tests/layout/test_array_layout.py [hunk 1]
```diff
@@ -424,6 +461,44 @@ def run(corsika_dict):
...
+    w_3 = {"corsika_sphere_radius": {"value": 16.0, "unit": "m"}}
+    t_3 = ArrayLayout._initialize_sphere_parameters(w_3)
+    assert pytest.approx(t_3["corsika_sphere_radius"].value, 0.1) == 16.0
+    assert t_3["corsika_sphere_radius"].unit == u.m
+
+    w_4 = {"corsika_sphere_radius": 16.0}
+    t_4 = ArrayLayout._initialize_sphere_parameters(w_4)
+    assert t_4 == {}
+
+    w_5 = {"_nocorsika_sphere_radius": 16.0}
```
Comment (orelgueta) [inline tests/unit_tests/layout/test_array_layout.py:498]: Should this have the unit in order to make this test a bit more. meaningful?

**Review by orelgueta — APPROVED** [review]
I approve with one small comment and despite the fact that reading this PR made me realise that we probably should have asked KB if he has a better name than "ground"...

File: tests/unit_tests/layout/test_array_layout.py (hunk 1 above)
Comment (GernotMaier) [reply tests/unit_tests/layout/test_array_layout.py]: So your comment actually made me realize that the test is wrong. The input to `_initialize_sphere_parameters` is of type `{LST: 12.5 m, MST: 9.15 m, SST: 3 m}`. Fixed this.

@orel - good you have another look at the test?

Comment (GernotMaier) [reply tests/unit_tests/layout/test_array_layout.py]: I must have had a reason for it, but all tests pass when adding an exception.

Could you have another look (sorry that it takes so many iterations).

Comment (orelgueta) [reply tests/unit_tests/layout/test_array_layout.py]: Oh that really confused me and cost 10 minutes. GitHub showed me changes only from some commits and not others I think and I saw mismatched test and code. 

It seems the test works now as expected (i.e., raises the exception), what else should be checked?

Comment (GernotMaier) [reply tests/unit_tests/layout/test_array_layout.py]: OK, not sure about the github issues.

All what changed is that an Exception is raised now, as you suggested. This has been added, plus the corresponding test, see: https://github.com/gammasim/simtools/pull/733/commits/37c5821aba76b293adf59c532ee27adbeb41950c

Comment (orelgueta) [reply tests/unit_tests/layout/test_array_layout.py]: Yes, I agree tests pass and all is good now. We can merge in my opinion.

**Pull request merged**